src/Conversion/ONNXToKrnl/Tensor/Reshape.cpp

```diff
@@ -12,16 +12,46 @@
 
 using namespace mlir;
 
+Value getDimOrConstant(ConversionPatternRewriter &rewriter, Location loc,
```

Comment:
This functionality stood go in the helper, if it does not exist already. In fact, IndexExpr has a lot of this things too, but I understand if you don't want to switch the implementation to that.

Reply:
@AlexandreEichenberger, thanks for your comments! I moved the function to a common place.

Here, we need to compute an output dim from values in a tensor. So we load a value from a tensor, and the value is always unknown. I am not sure it is appropriate to use IndexExpr here.